src/Interface/CMakeLists.txt

```diff
@@ -15,5 +19,7 @@ install(
   )
 
 if(HIOP_BUILD_SHARED)
-  add_library(hiopInterface OBJECT ${hiopInterface_SRC})
+  add_library(hiopInterface OBJECT ${hiopInterface_SRC} ${hiopInterface_PD})
+else()
+  add_library(hiopInterface OBJECT ${hiopInterface_PD})
 endif()
```

Comment:
I am not sure I understand this logic. @CameronRutherford, could you chime in here?

Reply:
chiopInterface.cpp is only needed when HIOP_BUILD_SHARED is on. For pridcomp algorithms we added a cpp file for the hiopInterfacePrimalDecomp class that needs to be added to the library at all time. But add_library can only be called once for one unique library name 'hiopInterface'. So I added the else there.

Reply:
I don't quite understand this either. Don't we just want to append `${hiopInterface_PD}` to `hiopInterface_SRC`?

Reply:
Sure. We can move the if statement to when hiopInterface_SRC is declared.

Reply:
Something along the lines of the following would also work:
```Cmake
add_library(hiopInterface OBJECT 
${hiopInterface_SRC}
$<$<BOOL:HIOP_BUILD_SHARED>:${hiopInterface_PD>)
```
I do think conditionally appending to `${hiopInterface_SRC}` is a better practice (the genexp can be much harder to read...), but @pelesh wanted me to chime in so I wanted to give another solution.

Reply:
OK. Since Cameron said better to add to hiopInterface_SRC, that's what I did. Thanks.